### ISSUE TITLE
Couldn't get type inference working

### ISSUE THREAD

**Issue by casey** [danburkert/prost#141]
I'm not sure if this is a bug, or if I'm not doing it right. From the readme, I thought that I could leave out annotations on fields and allow prost to infer the types. However, I got errors unless I added type annotations.

For example, this:

```
#[derive(Message)]
struct Foo {
  name: String,
}
```

Gave me the following error: 

```
error: proc-macro derive panicked
 --> src/foo.rs:3:10
  |
3 | #[derive(Message)]
  |          ^^^^^^^
  |
  = help: message: called `Result::unwrap()` on an `Err` value: ErrorMessage { msg: "no type attribute" }
          invalid message field Foo.name

error: aborting due to previous error
```

And adding an annotation got rid of the error:

```
#[derive(Message)]
struct Foo {
  #[prost(string)]
  name: String,
}
```

**Comment by danburkert** [comment]
`prost` has never supported this, and likely never will.  I think the confusion may stem from some issues in the README that were addressed in #365.  Please feel free to re-open if my assumption is off-base.

**Comment by danburkert** [comment]
To be clear, I'd love if type inference were a thing, but I don't know of a way to implement it.  Please feel free to open a separate issue if you have a proposal on how it could be done.
